driver/src/orderbook/streamed/balance.rs

```diff
@@ -60,137 +32,124 @@ impl Balance {
         // withdraw request because the smart contract should have emitted a withdraw event for the
         // previous one first.
         ensure!(
-            self.withdraw.batch_id >= current_batch_id || self.withdraw.amount == U256::zero(),
+            self.withdraw.batch_id >= current_batch_id || self.withdraw.amount.is_zero(),
             "new withdraw request before clearing of previous withdraw request"
         );
         self.withdraw.batch_id = batch_id;
-        self.withdraw.amount = amount;
+        self.withdraw.amount = bigint_u256::u256_to_bigint(amount);
```

Comment:
Wouldn't it be nice to implement from for this so that we can... `BigInt::from(amount)`